### ISSUE TITLE
vext breaks application logging

### ISSUE THREAD

**Issue by tbekolay** [stuaxo/vext#49]
I'm getting duplicated log entries outputted to the console when I have `vext` installed, and not when I don't have `vext` installed.

I believe something like this was brought up in #46; I'm not sure if this is the same issue or a different one, but it is still happening to me. The baffling thing to me is that the issue occurs if `vext` is installed in the current environment, even if `vext` isn't being imported or used in any way.

Here's a terminal session showing the issue:

```bash
~ mkvirtualenv tmp                                                                     
Running virtualenv with interpreter /usr/bin/python2
New python executable in /home/tbekolay/.virtualenvs/tmp/bin/python2
Also creating executable in /home/tbekolay/.virtualenvs/tmp/bin/python
Installing setuptools, pkg_resources, pip, wheel...done.
~ (tmp) python -c "import logging; print(logging.root.handlers)"                       
[]
~ (tmp) pip install vext                                                               
Collecting vext
Collecting ruamel.yaml>=0.11.10 (from vext)
  Downloading https://files.pythonhosted.org/packages/e8/1a/19687225f9a7a2bbbf61b74ed6e07e2e6bcc32f4dbaff01d8138ffb70c21/ruamel.yaml-0.15.43-cp27-cp27mu-manylinux1_x86_64.whl (585kB)
    100% |████████████████████████████████| 593kB 12.0MB/s 
Collecting ruamel.ordereddict; platform_python_implementation == "CPython" and python_version <= "2.7" (from ruamel.yaml>=0.11.10->vext)
  Using cached https://files.pythonhosted.org/packages/f3/2c/fa6d75dc459b371ed3b88fdbf8042785ce1655073c884fd97bdbb9f48e01/ruamel.ordereddict-0.4.13-cp27-cp27mu-manylinux1_x86_64.whl
Installing collected packages: ruamel.ordereddict, ruamel.yaml, vext
Successfully installed ruamel.ordereddict-0.4.13 ruamel.yaml-0.15.43 vext-0.7.0
~ (tmp) python -c "import logging; print(logging.root.handlers)"                       
[<logging.StreamHandler object at 0x7f2e9e21a110>]
```

As is shown above, in the fresh virtual environment, there are no handlers added to the root logger. All that's being done is importing `logging` and printing out the handlers.

After installing `vext`, we now get a `StreamHandler` defined seemingly at `python` startup, so that it is installed even if all we do is import `logging` and print out the handlers. Note that this a `StreamHandler` instance with log level 0 that printing messages to `stderr`; it is definitely not a `NullHandler`.

This happens in Python 2.7, 3.5 and 3.6 virtual environments.

**Comment by stuaxo** [comment]
The logging is setup here

https://github.com/stuaxo/vext/blob/3e033c7e67eb18f03b8770fc5e5c7157c0d50c8a/vext/__init__.py

It's worth experimenting with tweaking it there to see whats going on.

**Comment by mattrunchey** [comment]
I just filed #63 but didn't see this issue (sorry about possible duplicate). Pointed out what I believe fixes it there, though.

**Comment by ruanimal** [comment]
vext cause logging.basicConfig not work. 
The third-party library affects the global environment, which I think is very bad.

package info
```
vext==0.7.3
vext.gi==0.7.0
```
test code
```
import logging
import time
logging.basicConfig(level='DEBUG')
logger = logging.getLogger(__name__)
logger.info('aaa')
print(logging.root.manager.loggerDict)
```
output
```
{'vext': <Logger vext (WARNING)>, '__main__': <Logger __main__ (WARNING)>}
```

Correct output
```
INFO:__main__:aaa
{'__main__': <logging.Logger object at 0x10e8d6310>}
```

**Comment by stuaxo** [comment]
@ruanima can you see if the fix in #63 helps ?

**Comment by ruanimal** [comment]
> @ruanima can you see if the fix in #63 helps ?

I don’t think it will work. 

According to the documentation，`logging.basicConfig` must call before call log method like `debug()`, `info()`. 

`vext.install_importer()`   log debug message before user code,  which cause `logging.basicConfig `not work.

```python
def install_importer():
    logger.debug("install_importer has been moved to gatekeeper module")
    from vext import gatekeeper
    gatekeeper.install_importer()
```

![image](https://user-images.githubusercontent.com/14031394/85199205-3fdd1400-b320-11ea-9455-c099baa10433.png)

**Comment by stuaxo** [comment]
Good catch.

Do one of these two changes to the logging setup look better ?

The first version is less "weird", but then I don't know if adding a Null handler would be expected.

The second version goes out of it's way to avoid touching the logging framework unless VEXT_DEBUG_LOG is set.
```python
logger = logging.getLogger("vext")
if VEXT_DEBUG_LOG in environ and environ.get(VEXT_DEBUG_LOG, "0") == "1":
    logger.setLevel(logging.DEBUG)
else:
    logger.addHandler(logging.NullHandler())
```

```python
if VEXT_DEBUG_LOG in environ and environ.get(VEXT_DEBUG_LOG, "0") == "1":
    logger = logging.getLogger("vext")
    logger.setLevel(logging.DEBUG)
else:
    def logger(*args, **kwargs):
        """Fake logger, so we avoid touching configuring python logging at all"""
        pass
```

**Comment by ruanimal** [comment]
@stuaxo 

I think the first version will be better. Python documentation also advise to add NullHandler for library’s logger.

https://docs.python.org/3/howto/logging.html#configuring-logging-for-a-library

**Comment by stuaxo** [comment]
OK, I made a PR here
https://github.com/stuaxo/vext/pull/65

Let me know if that works for you.

**Comment by stuaxo** [comment]
Should I be adding a StreamHandler that outputs to console in the branch of the if statement ?

After 10 years of python, logging is still a little mysterious to me :)

**Comment by ruanimal** [comment]
@stuaxo 

I don't think it needs to add StreamHandler, because StreamHandler is the default log Handler，and the root logger defalut level is WARN.
If you want log message when developing library, you could config logging in your unittest.

This may be helpful for understanding logging.

![image](https://user-images.githubusercontent.com/14031394/85202986-0ebf0c80-b33d-11ea-80a4-787fd245302d.png)
